### ISSUE TITLE
How to train a STGCN with single GPU?

### ISSUE THREAD

**Issue by li624120638** [kennymckormick/pyskl#155]
Such as the title, My env only have single GPU

**Comment by kennymckormick** [comment]
You can use the following command (replace {num_gpus} with 1) to train with a single GPU. 
`bash tools/dist_train.sh {config_name} {num_gpus}` . 
Besides, if you don't change the batch size, you need to linearly decrease the learning rate when you decrease num_gpu.

**Comment by li624120638** [comment]
Thanks for your help. I have tried it, but it does not work. It might be because of my OS win10 which does not have `RANK` et. al variables. So, I try to annotation all code related to distributed training and allocate model and data from cpu to gpu, it works not bad.

**Comment by li624120638** [comment]
> You can use the following command (replace {num_gpus} with 1) to train with a single GPU. `bash tools/dist_train.sh {config_name} {num_gpus}` . Besides, if you don't change the batch size, you need to linearly decrease the learning rate when you decrease num_gpu.

Thanks for your help. I have tried it, but it does not work. It might be because of my OS win10 which does not have RANK et. al variables. So, I try to annotation all code related to distributed training and allocate model and data from cpu to gpu, it works not bad.

**Comment by kennymckormick** [comment]
Yes, we have not supported Windows training yet. But it's possible to do it with some modification (launch training directly with python, use a single GPU).

**Comment by xingxongxue** [comment]
Hello, I can run preprocessing on a single gpu windows computer, but the training times Error message Key Error: "RANK" or unrecognized arguments:1
What changes do I need to make to train on a windows computer

**Comment by kennymckormick** [comment]
The environment variable RANK is required when you are using distributed data parallel (DDP). To train on a Windows computer, you need to modify the code and remove all ddp parts and then launch the `train.py` directly with python, or you can modify the DDP arguments to make it compatible to Windows: (some reference) https://pytorch.org/tutorials/intermediate/ddp_tutorial.html.
